tsfc/mixedelement.py

```diff
@@ -95,7 +96,10 @@ def tabulate(self, order, points):
         crange = numpy.cumsum(sub_cmps)
 
         for i, e in enumerate(self.elements()):
-            table = e.tabulate(order, points)
+            try:
+                table = e.tabulate(order, points, entity)
+            except TraceError as TE:
+                table = TE.zeros
```

Comment:
In the case that a gradient of a trace element is tabulated, the exception `TE` will return the order=0 function evaluations as well as appropriately sized zeros in the dictionary for all orders of derivatives. That's what `TE.zeros` returns - it includes zeros for all gradient evaluations.